### PR TITLE
Add endReached & startReached

### LINKED ISSUE #437: App crash on large chunk of data on Android <= 12
**Describe the bug**
Hello there, I've encountered a bug where large chunks of data (250+) displayed in LineChart will crash the app instantly on Android devices lower than android 13. I've encountered the error in Android 12, 11, and 10. The app also doesn't show the linechart on Android 7, it stays empty. It does work as intended on Android 13 tho. (didn't try on other devices)

I have tried different versions of the package, 
- 1.3.16 
- 1.3.12 (which i was using originally)
- 1.3.2
- 1.2.42

**To Reproduce**
1. Send large chunks of data (over 250)
2. Open the app
3. Instant crash

I was able to reproduce the bug on a Snack project under Expo, so you can try it by yourself.
https://snack.expo.dev/@doctibbers/rn-gifted-charts-large-amount-bug?platform=android

It will work fine on any Android 13+ device and crashes on the rest.

**Expected behavior**
App should just show the normal data like the Android 13 behaviour

**Screenshots**
Screenshot_1700159526

 - Device: Any android device
 - OS: Android 12, 11, 10 (other versions also need to be tried)
 - Version of the library being used: 1.3.16, 1.3.12, 1.3.2, 1.2.42

### PR DESCRIPTION
**Pull request by hkellou-vitalcare**
I have tried to implement a feature that would allow users to trigger "endReached " and "startReached" functions in order to implement some sort of infinite scroll pagination on horizontal scrolls with a big load of data, on both BarChart and LineChart

This would technically bring some way to fix #437 

Please feel free to review my code. I have added an example for you in order to check the changes

### CODE REVIEW

File: package-lock.json
Comment (Abhinandan-Kushwaha) [inline package-lock.json:1]: Please don't push package-lock.json. Undo this change.

**Comment by Abhinandan-Kushwaha** [comment]
Thank you so much @hkellou-vitalcare for this valuable contribution. This feature will make the library more useful !!

**Comment by Abhinandan-Kushwaha** [comment]
I have added a few comments @hkellou-vitalcare 
Please resolve them.

Thanks once again for this PR!!

File: package-lock.json
Comment (hkellou-vitalcare) [reply package-lock.json:1]: @Abhinandan-Kushwaha I have reverted it back to master, apologies for this one

**Comment by hkellou-vitalcare** [comment]
I have reverted package-lock back to original and fixed a prop i did forget on the Wrapper. Should be fine now

**Comment by Abhinandan-Kushwaha** [comment]
I think you forgot to push the example files- **SimpleBarsEndReached** and **LineChartEndReached**

**Comment by hkellou-vitalcare** [comment]
That's correct. i think i didn't git add those files. Last commit should fix it

**Review by Abhinandan-Kushwaha — APPROVED** [review]

**Comment by Abhinandan-Kushwaha** [comment]
Hi @hkellou-vitalcare 
It is now available from version `1.4.1` onwards.
I have renamed the props to- `onStartReached` and `onEndReached`.
Also added an extra prop named `endReachedOffset`.

See [pagination related props](https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/blob/master/docs/BarChart/BarChartProps.md#pagination-related-props)
